src/routes/chat/[id]/+page.svelte

```diff
@@ -23,50 +23,33 @@
   let feed: HTMLOListElement;
   let input: HTMLInputElement;
 
-  onMount(() => {
-    invoke("get_usr_info", { bs58cert: peer_id}).then((info: any) => {
-      peername = info.username;
-      listen(`usr_name_${peer_id}`, (e: any) => {
-        peername = e.payload; 
-      });
-    });
+  onMount(async () => {
+    /* Setup the online user */
+    getUserInfo(peer_id).then((info) => { peername = info.name; });
+    onUserInfo(peer_id, (e) => { peername = e.name; });
 
-    invoke("get_local").then((user: any) => {
-      if (user === null) {
-        console.error("lol how did you manage to open a room without an identity")
-        localname = "[no_user_pem]"
-        return;
-      }
-      localname = user.info.username;
-      // Send our username to the peer.
-      emit(`send_message_${room_id}`, { Username: localname });
-
-      let local_id = user.identifier.bs58;
-      listen(`usr_name_${local_id}`, (e: any) => {
-        const name: string = e.payload;
-        console.log("localname changed");
-        localname = name
-        // Send our new username to the peer.
-        emit(`send_message_${room_id}`, { Username: localname });
-      });
-    });
+    /* Setup the local user */
+    const user = await getLocalUserInfo();
```

Comment:
collapse line 32 and 34 like line 28 for readability?

Reply:
Not sure what you mean by this, so I'll just leave it as is.
(I think it's plenty readable)